src/libcore/iter/iterator.rs

```diff
@@ -2463,6 +2463,52 @@ pub trait Iterator {
             }
         }
     }
+
+    /// Extends passed collection with the contents of an iterator.
+    ///
+    /// This method consumes the iterator and adds all its items to the
+    /// passed collection. The collection is then returned, so the call chain
+    /// can be continued.
+    ///
+    /// This method is a counter-part of `Extend::extend`, but instead of
+    /// being called on collection, it's called on iterator.
+    ///
+    /// ```
+    /// let result = (3..5).extend(vec![1, 2]);
+    /// assert_eq!(vec![1, 2, 3, 4], result);
+    /// ```
```

Comment:
Good example.. perhaps include a bit longer example as well which is more real-worldy?